File: C#/C# 2/1. Arrays-HW/Maximal-Sequence/MaximalSequence.cs
/*  Problem 4. Maximal sequence
    Write a program that finds the length of the maximal sequence of equal elements in an array of N integers.
    Input: 
    On the first line you will receive the number N
    On the next N lines the numbers of the array will be given
    Output: Print the length of the maximal sequence                 */

// Interpretation: The equal elements are consecutively placed.

using System;

class MaximalSequence
{
    static void Main()
    {
        //Input
        int n = int.Parse(Console.ReadLine()); 
        int[] arrayNums = new int[n];
        for (int i = 0; i < n; i++)
        {
            arrayNums[i] = int.Parse(Console.ReadLine());
        }

        int currentLength = 1;
        int maxLength = 0;
        //int elementBestLength = 0;

        for (int i = 0; i < arrayNums.Length - 1; i++)
        {
            if (arrayNums[i] == arrayNums[i + 1])
            {
                currentLength++;
            }
            else
            {
                if (currentLength > maxLength)
                {
                    maxLength = currentLength;
                    //elementBestLength = arrayNums[i];
                }
                currentLength = 1;
            }
        }

        // in case when the last sequence is the maximal sequence of equal elements
        if (currentLength > maxLength)
        {
            maxLength = currentLength;
            //elementBestLength = arrayNums[arrayNums.Length - 1];
        }

        Console.WriteLine(maxLength);

        //for (int i = 0; i < maxLength; i++)
        //{
        //    Console.Write(i != maxLength - 1 ? elementBestLength + ", " : elementBestLength + "\n");
        //}

    }
}

File: Front-end/JS Fundamentals/arrays/03. Maximal sequence/maximal-sequence.js
function solve(args) {
    var arr = args[0].split('\n'),
        n = +arr[0],
        numbers = arr.slice(1),  // copy the array from the first element 
        currentLength = 1,
        maxSeqLength = 0;

    for (i = 0; i < n - 1; i += 1) {
        if (+numbers[i] === +numbers[i + 1]) {
            currentLength += 1;
        } else {
            if (currentLength > maxSeqLength) {
                maxSeqLength = currentLength;
            }
            currentLength = 1;
        }
    }

    // in case when the last sequence is the maximal sequence of equal elements
    if (currentLength > maxSeqLength) {
        maxSeqLength = currentLength;
    }

    console.log(maxSeqLength);
}

solve(['10\n2\n1\n1\n2\n3\n3\n2\n2\n2\n1']);
